pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -34,7 +34,7 @@ type Store struct {
 type Getter interface {
 }
 
-func NewStore(cfg *config.CloudProviderCloudRunConfig, cloudProvider string, appLister applicationLister, logger *zap.Logger) *Store {
+func NewStore(_ context.Context, cfg *config.CloudProviderCloudRunConfig, cloudProvider string, appLister applicationLister, logger *zap.Logger) *Store {
```

Comment:
Tell me how to use ctx in brief?
I'm not sure your plan but basically letting struct have cotnext.Context is not a good manner.

Reply:
I guess it used to initialize the client of the store, not to be assigned as a property of the store. We could fix that on the caller side (on this PR: https://github.com/pipe-cd/pipecd/pull/3114#discussion_r791434181) if necessary.

Reply:
In that case, that makes sense. Just using for initial communication, right 👍

Reply:
Yep, you're right.
Therefore, it's assumed that client will be initialized in `NewStore` since I want the store to have a `client` instead of `ctx`